lib/contentTypeParser.js

```diff
@@ -161,14 +161,28 @@ ContentTypeParser.prototype.run = function (contentType, handler, request, reply
       done
     )
   } else {
-    const result = parser.fn(request, request[kRequestPayloadStream], done)
+    let result
+    try {
+      result = parser.fn(request, request[kRequestPayloadStream], done)
+    } catch (err) {
+      return done(err)
+    }
 
-    if (result && typeof result.then === 'function') {
-      result.then(body => done(null, body), done)
+    if (result) {
```

Comment:
This is not reliable at all. It could be always that the data to be serialized is returning an empty string, which is falsy.

```js
console.log('' ? true : false) // logs false
```

So this is totally unreliable. 

I dont even know what problem this PR is solving.

Reply:
> I dont even know what problem this PR is solving.

Problem is that when using a sync function to validate the Content Type, original implementation left the request in a pending status because both the callback was not being called and a Promise was not returned, and there's no info in the docs about it needing to be an async function at all. With my proposed change, it's possible to keep using a callback or return a Promise or return a validation / throw a validation error, so users can use whatever they want.

And yes, you are right that empty strings are falsy values, I'll update that conditional.

Reply:
See my usage in comment. It does not keep the usage of callback but complete remove the proper usage of callback.